extract.go

```diff
@@ -54,7 +56,18 @@ func Extract(c *cli.Context) {
 			fmt.Println("Extracting", entry.Name)
 		}
 
-		file, err := os.Create(path.Join(outdir, entry.Name))
+		outfile := path.Join(outdir, filepath.ToSlash(entry.Name))
+
+		if verbose {
+			fmt.Printf("Extract file path: %s", outfile)
+		}
+
+		err := os.MkdirAll(path.Dir(outfile), 0777)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Extract error: could not create folder %s.\nError: %s\n", path.Dir(outfile), err)
```

Comment:
Yes, I think it should. Reason I did not bail was to be coherent with the way file writing fails currently. Perhaps that could be changed in a different PR?